### PR TITLE
Add submodule: nlohmann json (Release 3.10.4)

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
 - *Add nlohmann_json (https://github.com/nlohmann/json) as submodule to parse json config files in benchmark_app and FrontEnds*

### Tickets:
 - *68214*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: thirdparty/CMakeLists.txt [hunk 1]
```diff
@@ -188,6 +188,14 @@ if(ENABLE_ONEDNN_FOR_GPU)
     build_onednn_gpu()
 endif()
 
+#
+# nlohmann json
+#
+if(ENABLE_SAMPLES OR NGRAPH_PDPD_FRONTEND_ENABLE OR NGRAPH_ONNX_FRONTEND_ENABLE OR NGRAPH_TF_FRONTEND_ENABLE)
```
Comment (ilyachur) [inline thirdparty/CMakeLists.txt:194]: Do we need this json library in samples?

File: .gitmodules [hunk 2]
```diff
@@ -59,3 +59,6 @@
 [submodule "tools/pot/thirdparty/open_model_zoo"]
 	path = tools/pot/thirdparty/open_model_zoo
 	url = https://github.com/openvinotoolkit/open_model_zoo.git
+[submodule "thirdparty/nlohmann_json"]
```
Comment (ilyachur) [inline .gitmodules:62]: @ttroilov FYI we are adding new dependency

File: thirdparty/CMakeLists.txt (hunk 1 above)
Comment (itikhono) [reply thirdparty/CMakeLists.txt:194]: yes, we are planning to use it in benchmark_app
previously we parsed json via OpenCV, but as far as I know, we abandon the dependency on this library.

**Review by ilyachur — APPROVED** [review]

**Review by ttroilov — APPROVED** [review]

**Comment by itikhono** [comment]
CI passed and we decided to merge it

**Pull request merged**